pydantic/main.py

```diff
@@ -211,6 +217,8 @@ def __new__(mcs, name, bases, namespace):
                         config=config,
                     )
 
+        if '__root__' in fields and len(fields) > 1:
+            raise ValueError('__root__ cannot be mixed with other fields')
```

Comment:
add `_custom_root_type: bool` to `new_namespace` below.

Reply:
I have added it.